lemminx-liberty/src/main/java/io/openliberty/tools/langserver/lemminx/util/XmlReader.java

```diff
@@ -0,0 +1,147 @@
+/*******************************************************************************
+* Copyright (c) 2023 IBM Corporation and others.
+*
+* This program and the accompanying materials are made available under the
+* terms of the Eclipse Public License v. 2.0 which is available at
+* http://www.eclipse.org/legal/epl-2.0.
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     IBM Corporation - initial API and implementation
+*******************************************************************************/
+
+package io.openliberty.tools.langserver.lemminx.util;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.logging.Logger;
+
+import javax.xml.stream.XMLEventReader;
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.events.XMLEvent;
+
+public class XmlReader {
+    private static final Logger LOGGER = Logger.getLogger(XmlReader.class.getName());
+
+    public static boolean hasServerRoot(File file) throws XMLStreamException, IOException {
+        if (!file.exists() || file.length() == 0) {
+            return false;
+        }
+        XMLInputFactory factory = XMLInputFactory.newInstance();
+        XMLEventReader reader = factory.createXMLEventReader(new FileInputStream(file));
+        if (reader.hasNext()) {
+            XMLEvent firstTag = reader.nextTag(); // first start/end element
+            reader.close();
+            return isServerElement(firstTag);
+        }
+        reader.close();
```

Comment:
So if something in this method throws an exception, won't the reader remain open? You should add a try/finally.